masq_lib/src/combined_parameters.rs

```diff
@@ -0,0 +1,647 @@
+            Self::RatePack(None) => Self::RatePack(Some(RatePack {
+                routing_byte_rate: CombinedParamsValueRetriever::get_value(
+                    &parsed_values,
+                    "routing_byte_rate",
+                ),
+                routing_service_rate: CombinedParamsValueRetriever::get_value(
+                    &parsed_values,
+                    "routing_service_rate",
+                ),
+                exit_byte_rate: CombinedParamsValueRetriever::get_value(
+                    &parsed_values,
+                    "exit_byte_rate",
+                ),
+                exit_service_rate: CombinedParamsValueRetriever::get_value(
+                    &parsed_values,
+                    "exit_service_rate",
+                ),
+            })),
+            Self::PaymentCurves(None) => Self::PaymentCurves(Some(PaymentCurves {
+                payment_suggested_after_sec: CombinedParamsValueRetriever::get_value(
+                    &parsed_values,
+                    "payment_suggested_after_sec",
+                ),
+                payment_grace_before_ban_sec: CombinedParamsValueRetriever::get_value(
+                    &parsed_values,
+                    "payment_grace_before_ban_sec",
+                ),
+                permanent_debt_allowed_gwei: CombinedParamsValueRetriever::get_value(
+                    &parsed_values,
+                    "permanent_debt_allowed_gwei",
+                ),
+                balance_to_decrease_from_gwei: CombinedParamsValueRetriever::get_value(
+                    &parsed_values,
+                    "balance_to_decrease_from_gwei",
+                ),
+                balance_decreases_for_sec: CombinedParamsValueRetriever::get_value(
+                    &parsed_values,
+                    "balance_decreases_for_sec",
+                ),
+                unban_when_balance_below_gwei: CombinedParamsValueRetriever::get_value(
+                    &parsed_values,
+                    "unban_when_balance_below_gwei",
+                ),
+            })),
+            Self::ScanIntervals(None) => Self::ScanIntervals(Some(ScanIntervals {
+                pending_payable_scan_interval: Duration::from_secs(
+                    CombinedParamsValueRetriever::get_value(
+                        &parsed_values,
+                        "pending_payable_scan_interval",
+                    ),
+                ),
+                payable_scan_interval: Duration::from_secs(
+                    CombinedParamsValueRetriever::get_value(
+                        &parsed_values,
+                        "payable_scan_interval",
+                    ),
+                ),
+                receivable_scan_interval: Duration::from_secs(
+                    CombinedParamsValueRetriever::get_value(
+                        &parsed_values,
+                        "receivable_scan_interval",
+                    ),
+                ),
+            })),
+            _ => panic!(
```

Comment:
Do we want to panic here or just write to error log what happened and why?

Reply:
This event indicates that the programmer's design is wrong; it should never happen so we want a panic to get to know as soon as possible from anyone seeing this panic. If so, it would require to change the code. So I think this is okay.